src/sentry/auth/access.py

```diff
@@ -127,10 +127,15 @@ def from_member(member):
     else:
         team_access = team_memberships
 
+    if scopes is not None:
+        scopes &= member.get_scopes()
```

Comment:
not sure if intentional or not but this will modify the object scopes points to in place and not just rebind it as a copy. This means that the caller gets its scopes modified.